src/base/net/stratum/SelfSelectClient.cpp

```diff
@@ -235,6 +240,49 @@ void xmrig::SelfSelectClient::submitBlockTemplate(rapidjson::Value &result)
     });
 }
 
+int64_t xmrig::SelfSelectClient::submit(const JobResult& result)
+{
+    if (m_submit_to_origin) {
+        submitOriginDaemon(result);
+    }
+    return m_client->submit(result);
+}
+
+void xmrig::SelfSelectClient::submitOriginDaemon(const JobResult& result)
+{
+    if (result.diff == 0) {
+        return;
+    }
+    
+    if (result.actualDiff() < m_targetdiff) {
+        m_origin_rejected++;
```

Comment:
Not submitted is better here, never sent

Reply:
Agreed